test/e2e/serial/tests/tolerations.go

```diff
@@ -433,6 +438,181 @@ var _ = Describe("[serial][disruptive][slow][rtetols] numaresources RTE tolerati
+					fxt.IsRebootTest = true
+					By("create NROP CR with no tolerations to the tainted node")
+					nropNewObj := nroOperObj.DeepCopy()
+					nropNewObj.ObjectMeta = metav1.ObjectMeta{
+						Name: nroOperObj.Name,
+					}
+					err := fxt.Client.Create(ctx, nropNewObj)
+					Expect(err).ToNot(HaveOccurred())
+
+					mcps, err := nropmcp.GetListByNodeGroupsV1(ctx, e2eclient.Client, nropNewObj.Spec.NodeGroups)
+					Expect(err).ToNot(HaveOccurred())
+					waitForMcpUpdate(fxt.Client, ctx, mcps)
+
+					klog.Info("waiting for DaemonSet to be ready")
+					ds, err := wait.With(fxt.Client).Interval(time.Second).Timeout(time.Minute).ForDaemonsetPodsCreation(ctx, dsKey, len(workers)-1)
+					Expect(err).NotTo(HaveOccurred(), "pods number is not as expected for daemonset: expected %d found %d", len(workers)-1, ds.Status.DesiredNumberScheduled)
+					_, err = wait.With(e2eclient.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(ctx, dsKey)
+					Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset %s: %v", dsKey.String(), err)
+
+					By("verify there is no RTE pod running on the tainted node")
+					pods, err := podlist.With(fxt.Client).ByDaemonset(ctx, *ds)
+					Expect(err).NotTo(HaveOccurred(), "Unable to get pods from Daemonset %q: %v", ds.Name, err)
+					Expect(len(pods)).To(Equal(len(workers)-1), "pods number is not as expected for daemonset %s/%s: expected %d found %d", ds.Namespace, ds.Name, len(workers)-1, len(pods))
+					for _, pod := range pods {
+						Expect(pod.Spec.NodeName).NotTo(Equal(taintedNode.Name), "found RTE pod running on tainted node without toleration on NROP obj")
+					}
+
+					By("add tolerations to NROP CR to tolerate the taint")
+					_ = setRTETolerations(ctx, fxt.Client, nroKey, testToleration())
+					klog.Info("waiting for DaemonSet pods to scale up and be ready")
+					_, err = wait.With(fxt.Client).Interval(time.Second).Timeout(time.Minute).ForDaemonsetPodsCreation(ctx, dsKey, len(workers))
+					Expect(err).NotTo(HaveOccurred(), "pods number is not as expected for daemonset: expected %d found %d", len(workers), ds.Status.DesiredNumberScheduled)
+					_, err = wait.With(fxt.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(ctx, dsKey)
+					Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset %s: %v", dsKey.String(), err)
+
+					By("verify there is a running RTE pod on the tainted node")
+					pods, err = podlist.With(fxt.Client).ByDaemonset(ctx, *ds)
+					Expect(err).NotTo(HaveOccurred(), "Unable to get pods from Daemonset %q: %v", ds.Name, err)
+					Expect(len(pods)).To(Equal(len(workers)), "pods number is not as expected for daemonset %s/%s: expected %d found %d", ds.Namespace, ds.Name, len(workers)-1, len(pods))
+					found := false
+					for _, pod := range pods {
+						if pod.Spec.NodeName == taintedNode.Name {
+							found = true
+							break
+						}
+					}
+					Expect(found).To(BeTrue())
```

Comment:
if we add a helper like suggested previously to find a RTE pod among all the pods running on a node (and I think there's something somewhere already, perhaps as a starting point? maybe u/s in the deployer or RTE repo), we can use the same code here to make sure such pod exists